kappa/context.py

```diff
@@ -23,39 +25,60 @@
 LOG = logging.getLogger(__name__)
 
 DebugFmtString = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
-InfoFmtString = '\t%(message)s'
+InfoFmtString = '...%(message)s'
 
 
 class Context(object):
 
-    def __init__(self, config_file, debug=False):
+    def __init__(self, config_file, environment=None,
+                 debug=False, force=False):
         if debug:
             self.set_logger('kappa', logging.DEBUG)
         else:
             self.set_logger('kappa', logging.INFO)
+        self._load_cache()
         self.config = yaml.load(config_file)
-        if 'policy' in self.config.get('iam', ''):
-            self.policy = kappa.policy.Policy(
-                self, self.config['iam']['policy'])
-        else:
-            self.policy = None
-        if 'role' in self.config.get('iam', ''):
-            self.role = kappa.role.Role(
-                self, self.config['iam']['role'])
-        else:
-            self.role = None
+        self.environment = environment
+        self.force = force
+        self.policy = kappa.policy.Policy(
+            self, self.config['environments'][self.environment])
+        self.role = kappa.role.Role(
+            self, self.config['environments'][self.environment])
         self.function = kappa.function.Function(
             self, self.config['lambda'])
         self.event_sources = []
         self._create_event_sources()
 
+    def _load_cache(self):
+        self.cache = {}
+        if os.path.isdir('.kappa'):
+            cache_file = os.path.join('.kappa', 'cache')
+            if os.path.isfile(cache_file):
+                with open(cache_file, 'rb') as fp:
+                    self.cache = yaml.load(fp)
```

Comment:
might want to do a utility function here, to use the c impl, and to use a SafeLoader, as default yaml encoder does arbitrary object serialization/de-serialization (aka pickle), which turns cache files into attack vectors. also is there a time limit on the cache?